src/preferences/preferences.cpp

```diff
@@ -1339,145 +1339,150 @@ void Preferences::disableRecursiveDownload(bool disable) {
 
 #ifdef Q_OS_WIN
 namespace {
-enum REG_SEARCH_TYPE {USER, SYSTEM_32BIT, SYSTEM_64BIT};
-
-QStringList getRegSubkeys(const HKEY &handle) {
-  QStringList keys;
-  DWORD subkeys_count = 0;
-  DWORD max_subkey_len = 0;
-  long res = ::RegQueryInfoKey(handle, NULL, NULL, NULL, &subkeys_count, &max_subkey_len, NULL, NULL, NULL, NULL, NULL, NULL);
-  if (res == ERROR_SUCCESS) {
-    max_subkey_len++; //For null character
-    LPTSTR key_name = new TCHAR[max_subkey_len];
-
-    for (uint i=0; i<subkeys_count; i++) {
-      res = ::RegEnumKeyEx(handle, 0, key_name, &max_subkey_len, NULL, NULL, NULL, NULL);
-      if (res == ERROR_SUCCESS)
-        keys.push_back(QString::fromWCharArray(key_name));
-    }
-    delete[] key_name;
-  }
+enum REG_SEARCH_TYPE
+{
+    USER,
+    SYSTEM_32BIT,
+    SYSTEM_64BIT
+};
 
-  return keys;
-}
+QStringList getRegSubkeys(HKEY handle)
+{
+    QStringList keys;
+
+    DWORD cSubKeys = 0;
+    DWORD cMaxSubKeyLen = 0;
+    LONG res = ::RegQueryInfoKeyW(handle, NULL, NULL, NULL, &cSubKeys, &cMaxSubKeyLen, NULL, NULL, NULL, NULL, NULL, NULL);
+
+    if (res == ERROR_SUCCESS) {
+        cMaxSubKeyLen++; // For null character
+        LPWSTR lpName = new WCHAR[cMaxSubKeyLen];
+        DWORD cName;
+
+        for (DWORD i = 0; i < cSubKeys; ++i) {
+            cName = cMaxSubKeyLen;
+            res = ::RegEnumKeyExW(handle, 0, lpName, &cName, NULL, NULL, NULL, NULL);
+            if (res == ERROR_SUCCESS)
+                keys.push_back(QString::fromWCharArray(lpName));
+        }
 
-QString getRegValue(const HKEY &handle, const QString &name = QString()) {
-  QString end_result;
-  DWORD type = 0;
-  DWORD size = 0;
-  DWORD array_size = 0;
+        delete[] lpName;
+    }
 
-  LPTSTR value_name = NULL;
-  if (!name.isEmpty()) {
-    value_name = new TCHAR[name.size()+1];
-    name.toWCharArray(value_name);
-    value_name[name.size()] = '\0';
-  }
+    return keys;
+}
 
-  // Discover the size of the value
-  ::RegQueryValueEx(handle, value_name, NULL, &type, NULL, &size);
-  array_size = size / sizeof(TCHAR);
-  if (size % sizeof(TCHAR))
-    array_size++;
```

Comment:
We have data size in ```size``` after previous call to RegQueryValueEx(). What makes you doubt that it is correct?..